ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala

```diff
@@ -8,17 +8,18 @@ import java.util.concurrent.TimeUnit
 import com.codahale.metrics
 import com.daml.ledger.participant.state.kvutils.DamlStateMap
 import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlConfigurationEntry,
   DamlLogEntry,
   DamlStateKey,
-  DamlStateValue,
-  DamlConfigurationEntry
+  DamlStateValue
 }
 import com.daml.ledger.participant.state.kvutils.{Conversions, Err}
 import com.daml.ledger.participant.state.v1.Configuration
 import com.digitalasset.daml.lf.data.InsertOrdMap
 import org.slf4j.Logger
 
 import scala.annotation.tailrec
+import scala.collection.immutable.SortedMap
 
 private[kvutils] object Common {
   type DamlOutputStateMap = Map[DamlStateKey, DamlStateValue]
```

Comment:
Why not make this a `SortedMap` rather than converting at the end?

Reply:
I'm wondering if the sortedness of keys is an implementation concern or not. If we say that the returned keys *will* be in a deterministic order then the returned type must be `SortedMap`, or a type that guarantees ordering otherwise the return type does not need to reflect that.

Reply:
That's fine, but either way, we could just build the map rather than converting it at the end. Or even use `SortedMap.newBuilder`.

Reply:
That's a good point -- will change construction of the map accordingly.